firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift

```diff
@@ -14,3 +14,14 @@ protocol AddressProvider {
 }
 
 extension RustAutofill: AddressProvider {}
+
+struct AddressLocaleFeatureValidator {
+    static let supportedRegions = ["CA", "US"]
+
+    static func isValidRegion() -> Bool {
+        guard let regionCode = Locale.current.regionCode else {
+            return false
+        }
+        return supportedRegions.contains(regionCode)
+    }
```

Comment:
nit: should we move this to a new file and maybe test? 🤣

Reply:
File is fine, but let me add tests. I can add them today although it feels wrong to push it so close to the deadline to we will do it for v129.1 

Going to first add tests and only then merge it in.

Reply:
cc @MattLichtenstein @cyndichin added some tests here